### PR TITLE
Reference Executable Semantics (Alternative)

### PR DESCRIPTION
**Pull request by bollu**
Alternate proposal to https://github.com/bollu/ssa/pull/36, with an aim of minimal change to `Core/`.

### CODE REVIEW

File: SSA/Projects/InstCombine/InstCombineBase.lean [hunk 1]
```diff
@@ -13,6 +13,10 @@ inductive BaseType
   | bitvec (w : Nat) : BaseType
   deriving DecidableEq
 
+instance : Repr BaseType where 
+  reprPrec 
+    | .bitvec w, _ => "bitvec<" ++ repr w ++">"
```
Comment (goens) [inline SSA/Projects/InstCombine/InstCombineBase.lean:18]: I would either go for the Lean version `"Bitvec " ++ repr w` or the MLIR version `"i" ++ repr w`, but this looks like some C++ inspired stuff
```suggestion
    | .bitvec w, _ => "i" ++ repr w ++">"
```

File: SSA/Projects/InstCombine/Tests.lean [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import SSA.Projects.InstCombine.InstCombineBase
+
+namespace InstCombine
+
+def tIdx (w : Nat) := (SSA.TSSAIndex.TERMINATOR (SSA.UserType.base (BaseType.bitvec w)))
+
+
```
Comment (goens) [inline SSA/Projects/InstCombine/Tests.lean:7]: I guess this is unnecessary now, right?
```suggestion

````

**Review by goens — COMMENTED** [review]
I like the API change, thanks!

**Comment by alexkeizer** [comment]
So far, this is only a way to execute specific tests that are pre-compiled into the binary, and not a way to execute arbitrary MLIR from a source file, right? Maybe we want to at least add an option for the second mode to the CLI, but stub out the implementation with an "Executing from a file is not supported yet" error.

**Comment by bollu** [comment]
@goens @alexkeizer another round of review / merge if you're happy!

**Pull request merged**